packages/lib/src/vis/hooks.ts

```diff
@@ -53,21 +53,34 @@ export function useDomains(
 
 export function useCameraState<T>(
   factory: (camera: Camera) => T,
-  equalityFn?: (prev: T, next: T) => boolean
+  deps: unknown[],
+  equalityFn = (prev: T, next: T) => Object.is(prev, next)
 ): T {
   const camera = useThree((state) => state.camera);
+  const rerender = useRerender();
 
-  const factoryRef = useSyncedRef(factory);
-  const [state, setState] = useState(() => factoryRef.current(camera));
+  const stateRef = useRef<T>(); // ref instead of state to avoid re-render when deps change
+  const factoryRef = useSyncedRef(factory); // ensure `useMemo` always sees latest `factory` reference
+
+  useMemo(() => {
+    // Compute state synchronously when deps change
+    stateRef.current = factoryRef.current(camera);
+  }, deps); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
No re-render here — the ref gets updated and returned synchronously.

Note that `useMemo` is important to avoid calling `factory` unnecessarily. Without it, we'd get: 

- re-render unrelated to camera or dependency change >> `factory` (unnecessary!)
- zoom >> `useFrame` >> `factory` >> re-render >> `factory` (again!)